### PR TITLE
Mod creative tab and stuff

### PR DESCRIPTION
**Pull request by PJWahlgren**
Also updated language for stuff

### CODE REVIEW

File: src/main/resources/assets/dysonsphereproject/lang/en_us.json [hunk 1]
```diff
@@ -6,5 +6,7 @@
   "tooltip.dysonspherecraft.solar_generator.tooltip": "Generates %1$s FE/t",
 
   "item.dysonsphereproject.mirror": "Mirror",
-  "item.dysonsphereproject.orbital_collector": "Orbital Collector"
+  "item.dysonsphereproject.orbital_collector": "Orbital Collector",
+
+  "creativetab.dsp_tab": "Dyson Sphere Project"
```
Comment (cmpashak) [inline src/main/resources/assets/dysonsphereproject/lang/en_us.json]: keep the naming convention - creativetab.dysonsphereproject.tab

**Review by cmpashak — CHANGES_REQUESTED** [review]

File: src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package com.tiph.dysonsphereproject.common.items;
```
Comment (zhouzhouthezhou) [inline src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java]: Does not belong in items, maybe move to init or common?

**Review by zhouzhouthezhou — CHANGES_REQUESTED** [review]

File: src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    throw new ClassCastException("Do not instansiate");
+  }
+
+  public static final DeferredRegister<CreativeModeTab> CREATIVE_MODE_TABS =
+      DeferredRegister.create(Registries.CREATIVE_MODE_TAB, DysonSphereProject.MODID);
+
+  public static void register(IEventBus eventBus) {
+    CREATIVE_MODE_TABS.register(eventBus);
+    CREATIVE_MODE_TABS.register(
+        "dsp_tab",
```
Comment (zhouzhouthezhou) [inline src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java]: same as connors review
dysonsphereproject.tab

File: src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        () ->
+            CreativeModeTab.builder()
+                .icon(() -> new ItemStack(DysonBlocks.SOLAR_GENERATOR))
+                .title(Component.translatable("creativetab.dsp_tab"))
```
Comment (zhouzhouthezhou) [inline src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java]: here too

File: src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                .displayItems(
+                    (param, output) -> {
+                      // BLOCKS
+                      output.accept(DysonBlocks.getBasicBlock(BasicBlocks.EXAMPLE_BASIC_BLOCK));
```
Comment (zhouzhouthezhou) [inline src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java]: while the hood is up make basic blocks loop through enum so we don't need to update everytime we add one

File: src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                      output.accept(DysonBlocks.SOLAR_GENERATOR);
+                      output.accept(DysonBlocks.WARP_DISLOCATOR);
+                      output.accept(DysonBlocks.GROUND_STATION);
+
+                      // ITEMS
+                      output.accept(DysonItems.getBasicItem(BasicItems.MIRROR));
```
Comment (zhouzhouthezhou) [inline src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java]: make basic items do the same

**Review by zhouzhouthezhou — CHANGES_REQUESTED** [review]

File: src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        "dsp_tab",
+        () ->
+            CreativeModeTab.builder()
+                .icon(() -> new ItemStack(DysonBlocks.SOLAR_GENERATOR))
+                .title(Component.translatable("creativetab.dsp_tab"))
+                .displayItems(
+                    (param, output) -> {
+                      // BLOCKS
+                      output.accept(DysonBlocks.getBasicBlock(BasicBlocks.EXAMPLE_BASIC_BLOCK));
+                      output.accept(DysonBlocks.SOLAR_GENERATOR);
```
Comment (zhouzhouthezhou) [inline src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java]: out of scope for this pr but we probably are are going to want to somehow automate this as well across all inits

File: src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java
Comment (zhouzhouthezhou) [inline src/main/java/com/tiph/dysonsphereproject/common/items/ModCreativeModeTabs.java:1]: Also please rename to DysonCreativeModeTabs

**Review by zhouzhouthezhou — CHANGES_REQUESTED** [review]

File: src/main/java/com/tiph/dysonsphereproject/common/init/ModCreativeModeTabs.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.tiph.dysonsphereproject.DysonSphereProject;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.CreativeModeTab;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.IEventBus;
+import net.neoforged.neoforge.registries.DeferredRegister;
+
+public class ModCreativeModeTabs {
```
Comment (cmpashak) [inline src/main/java/com/tiph/dysonsphereproject/common/init/ModCreativeModeTabs.java]: rename this to DysonCreativeModeTab

**Review by cmpashak — CHANGES_REQUESTED** [review]

**Review by cmpashak — APPROVED** [review]

**Pull request merged**